python/lsst/meas/base/forcedPhotCcd.py

```diff
@@ -323,36 +392,161 @@ def filterReferences(self, exposure, refCat, refWcs):
+        #     refRecord.setFootprint(refRecord.getFootprint().transform(refWcs,
+        #                                                               expWcs, expRegion))
+        # # Step 5: Replace reference catalog with filtered source list.
+        # return refSources
+
+    def generateMeasCat(self, exposureDataId, exposure, refCat, refWcs, idPackerName):
+        """Generate a measurement catalog for Gen3.
+
+        Parameters
+        ----------
+        exposureDataId : `DataId`
+            Butler dataId for this exposure.
+        exposure : `lsst.afw.image.exposure.Exposure`
+            Exposure to generate the catalog for.
+        refCat : `lsst.afw.table.SourceCatalog`
+            Catalog of shapes and positions at which to force photometry.
+        refWcs : `lsst.afw.image.SkyWcs`
+            Reference world coordinate system.
+        idPackerName : `str`
+            Type of ID packer to construct from the registry.
+
+        Returns
+        -------
+        measCat : `lsst.afw.table.SourceCatalog`
+            Catalog of forced sources to measure.
+        expId : `int`
+            Unique binary id associated with the input exposure
+        """
+        expId, expBits = exposureDataId.pack(idPackerName, returnMaxBits=True)
+        idFactory = lsst.afw.table.IdFactory.makeSource(expId, 64 - expBits)
+
+        measCat = self.measurement.generateMeasCat(exposure, refCat, refWcs,
+                                                   idFactory=idFactory)
+        return measCat, expId
+
+    def runDataRef(self, dataRef, psfCache=None):
+        """Perform forced measurement on a single exposure.
+
+        Parameters
+        ----------
+        dataRef : `lsst.daf.persistence.ButlerDataRef`
+            Passed to the ``references`` subtask to obtain the reference WCS,
+            the ``getExposure`` method (implemented by derived classes) to
+            read the measurment image, and the ``fetchReferences`` method to
+            get the exposure and load the reference catalog (see
+            :lsst-task`lsst.meas.base.references.CoaddSrcReferencesTask`).
+            Refer to derived class documentation for details of the datasets
+            and data ID keys which are used.
+        psfCache : `int`, optional
+            Size of PSF cache, or `None`. The size of the PSF cache can have
+            a significant effect upon the runtime for complicated PSF models.
+
+        Notes
+        -----
+        Sources are generated with ``generateMeasCat`` in the ``measurement``
+        subtask. These are passed to ``measurement``'s ``run`` method, which
+        fills the source catalog with the forced measurement results. The
+        sources are then passed to the ``writeOutputs`` method (implemented by
+        derived classes) which writes the outputs.
+        """
+        refWcs = self.references.getWcs(dataRef)
+        exposure = self.getExposure(dataRef)
+        if psfCache is not None:
```

Comment:
It should be, but it would be done in a different way to this, with some kind of config attribute or something. However, I feel that is outside the scope of this ticket, which was just to orthogonalize the two tasks and make the task work with overlapping patches.